### DISCUSSION TITLE
Format column on PivotTable Sheet & correctly set column range for PivotTable

### DISCUSSION THREAD

**Discussion by MarianiVentura** [dfinke/ImportExcel#1445] [General]
Hello,

Would you be able to help me with two things.

Consider my code:

```
$excel = Import-Csv $output_file -Delimiter ^ | Export-Excel $excel_file -AutoSize -AutoNameRange -TableName Items -FreezePane 3 -Passthru `
-IncludePivotTable  -pivotrows Days -PivotData @{BOvalue='sum'} -IncludePivotChart  -ChartType Pie  -ShowPercent -show
$sheet = $excel.Workbook.Worksheets["Sheet1"]
$sheet.Column(8) | Set-ExcelRange -NFormat "$#,###.##"
$sheet.Column(9) | Set-ExcelRange -NFormat "$#,###.##"
$sheet.Column(12) | Set-ExcelRange -NFormat "$###,###,###.##"
$sheet.Column(13) | Set-ExcelRange -NFormat "$###,###,###.##"
$sheet.Column(14) | Set-ExcelRange -NFormat "$###,###,###.##"
#$sheet2 = $excel.Workbook.Worksheets["Sheet1PivotTable"]
#$sheet2.Column(2) | Set-ExcelRange -NFormat "$###,###,###.##"
Close-ExcelPackage -ExcelPackage $excel 
```
I would like to learn how to set a range for my pivot table that only includes a range.  I have a summary formula at the top of the "BOvalue" column and I do not want to include the summary row in the pivot table data.

Also, I successfully formatted some columns on Sheet1 to format as numbers with commas and periods, but I wasn't able to do this for the second column on my new "Sheet1PivotTable" sheet.  The two lines are currently remarked out in my code.  

Thanks for any assistance, much appreciated!

M.

**Comment by dfinke** [comment]
pls could you post the data your are using and the xlsx of how you would like it to look

**Comment by MarianiVentura** [comment]
Absolutely!  Thanks for the assistance.
![image](https://user-images.githubusercontent.com/21973914/233844746-cf4f2696-5b64-448e-902a-463ae4236b02.png)
![image](https://user-images.githubusercontent.com/21973914/233844756-351a1300-b3c2-4d99-b198-a393f1ffe3fb.png)

**Comment by MarianiVentura** [comment]
Good day.  Here is my [ExcelFile](https://accentdecor-my.sharepoint.com/:x:/p/michael_mariani/EdCHCc0wwR9Khc5NcGXXRi8Be3ipi3RifVNnOfbg3AZUDA?e=TXHmdp)

I had to manually add the totals on row 1.  If I don't, they become part of the table and the pivot table calculations.
Is there a way in Powershell to insert this line?  Or specify a range for the table?
Also, How do I format the values in my pivot table?  We want to show the commas at least and it would be nice to have $.  I can do this on columns on the main sheet, but they are not formatted on the pivot table.  
Thanks!
